sklearn/isotonic.py

```diff
@@ -413,8 +419,9 @@ def __setstate__(self, state):
         We need to rebuild the interpolation function.
         """
         super().__setstate__(state)
-        if hasattr(self, '_necessary_X_') and hasattr(self, '_necessary_y_'):
-            self._build_f(self._necessary_X_, self._necessary_y_)
```

Comment:
I think it would be good to add a bit of backward compat code here to make the effort to keep it possible to unpickle `IsotonicRegression` instances that used the old private attributes.